### DISCUSSION TITLE
Jekyll redirects to a 404 and does not render post.md

### DISCUSSION THREAD

**Discussion by jtmlis** [community/community#23449] [Other Feature Feedback, Questions, & Ideas]
<p>I want my index.md seen at</p>
<ul>
<li><a href="http://jtm-lis.github.io/Julien_Tremblay_McLellan/" rel="nofollow noopener">jtm-lis.github.io/Julien_Tremblay_McLellan/</a></li>
</ul>
<p>to redirect to pages I have written in markdown</p>
<p>I tried implementing, post_url variable , such as documented<br>
in order to successful redirect to a page written in markdown without success, as it points to a 404 at</p>
<pre><code class="lang-auto">[Name of Link]({% post_url 2010-07-21-name-of-post %})

</code></pre>
<p>Image taken from  <a href="http://jekyllrb.com/docs/liquid/tags/#linking-to-posts" rel="nofollow noopener">jekyllrb.com/docs/liquid/tags/#linking-to-posts</a><br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/696135265831a8b5ed00793ef893721699336174.png" data-download-href="/uploads/short-url/f2eoQv3jdLL2jePoKlwwWJN3pKk.png?dl=1" title="" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/66714119/181077153-6560aa69-129b-41ee-8c0c-046c196b70c6.png" alt="" data-base62-sha1="f2eoQv3jdLL2jePoKlwwWJN3pKk" width="690" height="482" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/696135265831a8b5ed00793ef893721699336174_2_690x482.png, https://aws1.discourse-cdn.com/github/original/2X/6/696135265831a8b5ed00793ef893721699336174.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/6/696135265831a8b5ed00793ef893721699336174.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/696135265831a8b5ed00793ef893721699336174_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename"></span><span class="informations">849×594 84.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>At first I thought this was an error from the for loop, so I added the link manually as detailed in the documentation specifically for pages written in markdown.</p>
<h2>index.md</h2>
<pre><code class="lang-auto"># Index of all my content

[Library Carpentry Workshop July 2020]({% post_url 2020-07-27-library-carpentry-workshop-american-university-notes %})

&lt;ul&gt;
  {% for post in site.posts %}
    &lt;li&gt;
      &lt;a href="{{ post.url }}"&gt;{{ post.title }}&lt;/a&gt;
    &lt;/li&gt;
  {% endfor %}
&lt;/ul&gt;
</code></pre>
<h2>_config.yml</h2>
<pre><code class="lang-auto">theme: jekyll-theme-slate
url: https://jtm-lis.github.io
baseurl: /Julien_Tremblay_McLellan #NO TRAILING SLASH
title: Julien Tremblay McLellan's Website
author: Julien Tremblay McLellan
email: jtremc@gmail.com
description: &gt; # this means to ignore newlines until "baseurl:"
  Write an awesome description for your new site here. You can edit this
  line in _config.yml. It will appear in your document head meta (for
  Google search results) and in your feed.xml site description.
# social links
twitter_username: jtm-lis # DO NOT include the @ character, or else the build will fail!
github_username:  jtm-lis # DO NOT include the @ character, or else the build will fail!

show_excerpts: true # set to false to remove excerpts on the homepage

</code></pre>
<p>New users can only post 2 links.</p>
<p>What am I doing wrong?</p>

**Comment by otter-computer** [comment, marked as answer]
<p>It looks like the links that are being generated from this code are missing the repository name baseurl <code>Julien_Tremblay_McLellan/</code> from the URL.</p>
<p>e.g. the link to your site is:</p>
<p><code>https://jtm-lis.github.io/Julien_Tremblay_McLellan/</code></p>
<p>However, the URLs that are being generated look like the following:</p>
<p><code>https://jtm-lis.github.io/2020/07/27/library-carpentry-workshop-american-university-notes.html</code></p>
<p>This results in a 404 as it’s missing the repository name <code>Julien_Tremblay_McLellan/</code>, which is the <code>baseurl</code> value, from the URL.</p>
<p><em>Note that the Jekyll website mentions that you don’t need to do this as of Jekyll version 4.0. However, <a href="https://pages.github.com/versions/" rel="nofollow noopener">GitHub Pages still runs Jekyll 3.8.7</a> underneath so you will need to manually include the <code>baseurl</code>.</em></p>
<p>There are two ways you can fix this. The first—and probably the best, if this is a personal blogging website—is to turn this Pages site into a <a href="https://docs.github.com/en/github/working-with-github-pages/about-github-pages#types-of-github-pages-sites" rel="nofollow noopener">User Pages site</a>. To do this you’ll need to rename your repository from <code>Julien_Tremblay_McLellan</code> to <code>jtm-lis.github.io</code>. This will change your site’s URL to <a href="https://jtm-lis.github.io/" rel="nofollow noopener">https://jtm-lis.github.io/</a> and these generated links should now work.</p>
<p>Alternatively you can modify the code to automatically include the <code>baseurl</code> when generating these links. The easiest way to do this would be to turn these links into a relative URL:</p>
<pre><code class="lang-auto">&lt;ul&gt;
  {% for post in site.posts %}
    &lt;li&gt;
      &lt;a href="{{ post.url | relative_url }}"&gt;{{ post.title }}&lt;/a&gt;
    &lt;/li&gt;
  {% endfor %}
&lt;/ul&gt;
</code></pre>
<p>If you’re going to use the first method I’d probably recommend you <em>also</em> do the second too, but both should do the job!</p>

**Reply by chancej715** [reply]
Thanks!!

**Comment by jtmlis** [comment]
<p>Thank you so much <a class="mention" href="/thomasshaped">@thomasshaped</a> !! It worked.</p>

**Comment by fredericpierron** [comment]
<p>Hi,<br>
I did another trick: I’ve just changed in the _config.yml the url from</p>
<p>url:<a href="https://blabla.github.io" rel="noopener nofollow ugc">https://blabla.github.io</a></p>
<p>to</p>
<p>url:<a href="https://blabla.github.io/the_needed_part" rel="noopener nofollow ugc">https://blabla.github.io/the_needed_part</a></p>
